### DISCUSSION TITLE
Problems with libharfbuzz.so.0 on linux

### DISCUSSION THREAD

**Discussion by MilkPiggies** [KevinBytesTheDust/CVAmp#321] [Q&A] [answered]
Please bare with me as this is my first time posting a discussion on Github.
I am not sure if this should be posted as an issue or a question. 

I have recently had to reinstall linux, and in doing so, I seem to have gotten rid of a dependency that CVAmp uses. I have researched it for the last day, and have found several things it could have been, but nothing seems to fix it.

From my understanding the issue seems to be with: 
/opt/google/chrome/chrome/libharfbuzz.so.0: undefined symbol: FT_Get_Transform

Here is my cvamp.log:
2024-12-21 20:57:40,734;INFO;31_26;MainThread;manager;__init__;Manager start on Linux-6.8.0-51-generic-x86_64-with-glibc2.39
2024-12-21 20:57:43,214;INFO;0_27;Thread-2 (spawn_instance_thread);manager;spawn_instance_thread;Ordered TWITCH instance 1, Thread-2 (spawn_instance_thread), proxy http://<ip-blocked-for-privacy>
2024-12-21 20:57:43,216;INFO;100_27;Thread-2 (spawn_instance_thread);manager;instance_status_report_callback;STARTING instance 1
2024-12-21 20:57:43,678;ERROR;47_27;Thread-2 (spawn_instance_thread);instance;start;Target page, context or browser has been closed
Browser logs:

<launching> /opt/google/chrome/chrome --disable-field-trial-config --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-back-forward-cache --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-component-update --no-default-browser-check --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose,MediaRouter,DialMediaRouteProvider,AcceptCHFrame,AutoExpandDetailsElement,CertificateTransparencyComponentUpdater,AvoidUnnecessaryBeforeUnloadCheckSync,Translate,HttpsUpgrades,PaintHolding --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --disable-search-engine-choice-screen --headless --hide-scrollbars --mute-audio --blink-settings=primaryHoverType=2,availableHoverTypes=2,primaryPointerType=4,availablePointerTypes=4 --no-sandbox --proxy-server=http://<ip-blocked-for-privacy> --proxy-bypass-list=<-loopback> --window-position=0,0 --mute-audio --user-data-dir=/tmp/playwright_chromiumdev_profile-xdOY5C --remote-debugging-pipe --no-startup-window
<launched> pid=482288
[pid=482288][err] /opt/google/chrome/chrome: symbol lookup error: /lib/x86_64-linux-gnu/libharfbuzz.so.0: undefined symbol: FT_Get_Transform died at page None
Traceback (most recent call last):
  File "cvamp/instance.py", line 85, in start
  File "cvamp/instance.py", line 136, in spawn_page
  File "playwright/sync_api/_generated.py", line 14882, in launch
  File "playwright/_impl/_sync_base.py", line 115, in _sync
  File "playwright/_impl/_browser_type.py", line 94, in launch
  File "playwright/_impl/_connection.py", line 63, in send
  File "playwright/_impl/_connection.py", line 495, in wrap_api_call
  File "playwright/_impl/_connection.py", line 101, in inner_send
playwright._impl._errors.TargetClosedError: Target page, context or browser has been closed
Browser logs:

<launching> /opt/google/chrome/chrome --disable-field-trial-config --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-back-forward-cache --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-component-update --no-default-browser-check --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose,MediaRouter,DialMediaRouteProvider,AcceptCHFrame,AutoExpandDetailsElement,CertificateTransparencyComponentUpdater,AvoidUnnecessaryBeforeUnloadCheckSync,Translate,HttpsUpgrades,PaintHolding --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --disable-search-engine-choice-screen --headless --hide-scrollbars --mute-audio --blink-settings=primaryHoverType=2,availableHoverTypes=2,primaryPointerType=4,availablePointerTypes=4 --no-sandbox --proxy-server=<ip-blocked-for-privacy> --proxy-bypass-list=<-loopback> --window-position=0,0 --mute-audio --user-data-dir=/tmp/playwright_chromiumdev_profile-xdOY5C --remote-debugging-pipe --no-startup-window
<launched> pid=482288
[pid=482288][err] /opt/google/chrome/chrome: symbol lookup error: /lib/x86_64-linux-gnu/libharfbuzz.so.0: undefined symbol: FT_Get_Transform
2024-12-21 20:57:43,682;INFO;100_27;Thread-2 (spawn_instance_thread);manager;instance_status_report_callback;SHUTDOWN instance 1

If anyone could help me find a fix, or point me in the right direction, it would be greatly appreciated.

I have tried the latest release of CVAmp and the one before it, to no avail. I have also tried reinstalling libharfbuzz6 and freetype2 (from what I read these were possible fixes). If there's any more information needed, please let me know!

Thank you in advance.

**Comment by KevinBytesTheDust** [comment, marked as answer]
See here: https://github.com/KevinBytesTheDust/CVAmp/issues/326
